uservalidation/user_test.go

```diff
@@ -6,69 +6,54 @@ import (
 )
 
 func TestValidate(t *testing.T) {
-	type args struct {
-		user     User
-		isSignUp bool
-	}
 	tests := []struct {
-		name    string
-		args    args
-		wantErr bool
+		name          string
+		inputUser     User
+		inputIsSignUp bool
+		outputErr     bool
 	}{
 		{
 			"Invalid first name",
-			args{
-				User{"J", "Doe", "john@doe.com", "Password123!"},
-				true,
-			},
+			User{"J", "Doe", "john@doe.com", "Password123!"},
+			true,
 			true,
 		},
 		{
 			"Invalid last name",
-			args{
-				User{"John", "D", "john@doe.com", "Password123!"},
-				true,
-			},
+			User{"John", "D", "john@doe.com", "Password123!"},
+			true,
 			true,
 		},
 
 		{
 			"Invalid email",
-			args{
-				User{"John", "Doe", "johndoe.com", "Password123!"},
-				true,
-			},
+			User{"John", "Doe", "johndoe.com", "Password123!"},
+			true,
 			true,
 		},
 		{
 			"Invalid password",
-			args{
-				User{"John", "Doe", "john@doe.com", "123"},
-				true,
-			},
+			User{"John", "Doe", "john@doe.com", "123"},
+			true,
 			true,
 		},
 		{
 			"Valid User - sign up",
-			args{
-				User{"John", "Doe", "john@doe.com", "Password123!"},
-				true,
-			},
+			User{"John", "Doe", "john@doe.com", "Password123!"},
+			true,
 			false,
 		},
 		{
 			"Valid User - login",
-			args{
-				User{"", "", "john@doe.com", "Password123!"},
-				false,
-			},
+			User{"", "", "john@doe.com", "Password123!"},
+			false,
 			false,
 		},
 	}
 	for _, tt := range tests {
```

Comment:
Normalmente variáveis com duas letras repetidas inferem coleções em Go. Como tá sendo usada pra inferir um único teste aqui pode gerar confusão.

Uma forma de resolver seria a abaixo, levando em consideração que em testes a variável `t` já é usada.

```suggestion
	for _, tc := range testCases {
```